compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -298,11 +298,18 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                             .get(0)
                             .map(|p| (p.span.shrink_to_lo(), "&self, "))
                             .unwrap_or_else(|| {
+                                // Try to look for the "(" after the function name, if possible.
+                                // This avoids placing the suggestion into the visibility specifier.
+                                let span = if let Some(fn_ident) = fn_kind.ident() {
+                                    span.trim_start(fn_ident.span).unwrap_or(*span)
+                                } else {
+                                    *span
+                                };
```

Comment:
I think you can write this as:

```suggestion
                                let span = fn_kind.ident().map_or(*span, |ident| span.with_lo(ident.span.hi()));
```

(you'll have to run `./x.py fmt` again)

Reply:
A lot cleaner, thanks :)